File: C/C++/114.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main(){
    int N;
    cin >> N;
    vector<string> ls(N);
    for (int i = 0; i < N; i++){
        cin >> ls[i];
    }

    for (int i = 0; i < N - 1; i++){
        char lastChar = ls[i].back();
        char firstChar = ls[i+1][0];

        if (lastChar == firstChar){
            continue;
        }else{
            cout << lastChar << " " << firstChar << endl;
            return 0;
        }
    }
    cout << "Yes" << endl;
    return 0;
}

File: C/Java/114.java
import java.util.*;

public class Main{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        sc.nextLine();
        String[] ls = new String[N];
        for (int i = 0; i < N; i++){
            ls[i] = sc.nextLine();
        }

        for (int i = 0; i < N-1; i++){
            char lastChar = ls[i].charAt(ls[i].length()-1);
            char firstChar = ls[i+1].charAt(0);

            if (lastChar == firstChar){
                continue;
            }else{
                System.out.println(lastChar + " " + firstChar);
                return;
            }
        }
        System.out.println("Yes");
    }
}